fs/fsx.py

```diff
@@ -46,63 +58,86 @@ def setup_tmpfs_dir(self):
-            'https://raw.githubusercontent.com/linux-test-project/ltp/'
-            'master/testcases/kernel/fs/fsx-linux/fsx-linux.c', expire='7d')
-        os.chdir(self.workdir)
-        process.system('gcc -o fsx %s' % fsx, shell=True, ignore_status=True)
+        self.thp_page_cache = self.params.get('thp_page_cache', default=False)
+        self.disk = self.params.get('disk', default="none")
+        if self.thp_page_cache:
+            self.dir = self.params.get('dir', default=self.workdir)
+            if self.dir:
+                self.setup_tmpfs_dir()
+                self.output = os.path.join(self.dir, 'result')
+            else:
+                self.cancel("dir not specified")
+        else:
+            self.output = self.params.get('output_file', default=None)
+            if not self.output:
+                self.output=os.path.join(self.workdir,"result")
+
+        if not os.path.exists(self.output):
+            os.makedirs(self.output)
+
+        url = self.params.get(
+            'url', default='https://github.com/linux-test-project/ltp/archive/master.zip')
+        match = next((ext for ext in [".zip", ".tar"] if ext in url), None)
+        tarball = ''
+        if match:
+            tarball = self.fetch_asset(
+                "ltp-master%s" % match, locations=[url], expire='7d')
+        else:
+            self.cancel("Provided LTP Url is not valid")
+        self.ltpdir_parent = os.path.join(self.workdir,'/ltp')
+        if not os.path.exists(self.ltpdir_parent):
+            os.mkdir(self.ltpdir_parent)
+        archive.extract(tarball, self.ltpdir_parent)
+        self.ltp_dir = os.path.join(self.ltpdir_parent, "ltp-master")
+        os.chdir(self.ltp_dir)
+        build.make(self.ltp_dir, extra_args='autotools')
+        process.system('./configure')
         self.thp = False
 
     def test(self):
         '''
         Run Fsx test for exercising file system
         '''
-        file_ub = self.params.get('file_ub', default='1000000')
-        op_ub = self.params.get('op_ub', default='1000000')
-        output = self.params.get('output_file', default='/tmp/result')
-        num_times = self.params.get('num_times', default='10000')
-        self.mount_dir = self.params.get('tmpfs_mount_dir', default=None)
-        thp_page_cache = self.params.get('thp_page_cache', default=None)
-
-        if thp_page_cache:
-            if self.mount_dir:
-                self.setup_tmpfs_dir()
-                output = os.path.join(self.mount_dir, 'result')
-            else:
-                self.cancel("tmpfs_mount_dir not specified")
-        else:
-            output = self.params.get('output_file', default='/tmp/result')
+        test_file_max_size = self.params.get('test_file_max_size', default='1000000')
+        single_op_max_size = self.params.get('single_op_max_size', default='1000000')
+        total_ops = self.params.get('total_ops', default='1000')
+        num_times = self.params.get('num_times', default='1')
```

Comment:
@AYUSHJAIN951 sorry for last comments, can we move all `self.params.get` to setup?
rest looks good to me.


Regards
Naresh

Reply:
can you review again @Naresh-ibm